### PR TITLE
test: enabe back archiving experiment action in kfp_functional_test

### PR DESCRIPTION
**Pull request by difince**
**Description of your changes:**
Enable back archiving experiment action in kfp_functional_test and upgrade to using kfp.v2 

Resolves: https://github.com/kubeflow/pipelines/issues/9770

Depends on PR: https://github.com/kubeflow/pipelines/pull/9730

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: test/kfp-functional-test/run_kfp_functional_test.py
Comment (gkcalat) [inline test/kfp-functional-test/run_kfp_functional_test.py:1]: It seems to me that the functional testing in [broken](https://oss.gprow.dev/view/gs/oss-prow/logs/kubeflow-pipelines-periodic-functional-test/1681338338910408704) due to namespace changes in the SDK. Would you be willing to extend the scope of this PR to upgrading `kfp` SDK to v2?

Comment (difince) [reply test/kfp-functional-test/run_kfp_functional_test.py:1]: done

Comment (difince) [reply test/kfp-functional-test/run_kfp_functional_test.py:1]: @gkcalat Please review my latest changes - I updated the `kfp_functional_test` to be compatible with KFP version 2.0.1. 
I have tested the script locally and it works but for authentication, I have used cookie `authservice_session` :
Below I am providing the code snipped I used for authentication. 

```
HOST = "http://localhost:8080/"
USERNAME = "user@example.com"
PASSWORD = "12341234"
USER_NAMESPACE = "kubeflow-user-example-com"

session = requests.Session()
response = session.get(HOST)

headers = {
    "Content-Type": "application/x-www-form-urlencoded",
}

data = {"login": USERNAME, "password": PASSWORD}
session.post(response.url, headers=headers, data=data)
print(session.cookies.get_dict())
session_cookie = session.cookies.get_dict()["authservice_session"]
client = kfp.Client(
    host=f"{HOST}/pipeline",
    cookies=f"authservice_session={session_cookie}",
    namespace=USER_NAMESPACE,
)
```
So, IMO the script should work fine against ` --host "https://$(curl https://raw.githubusercontent.com/kubeflow/testing/master/test-infra/kfp/endpoint)"`, if I am able to successfully authorize using `gcloud auth application-default login` ..

Comment (gkcalat) [reply test/kfp-functional-test/run_kfp_functional_test.py:1]: I was able to validate the test with archiving experiment after these additional changes: https://github.com/difince/pipelines/pull/8.

Comment (difince) [reply test/kfp-functional-test/run_kfp_functional_test.py:1]: @gkcalat your changes have been merged! Thank you

**Comment by gkcalat** [comment]
Sample tests failure is unrelated to this change.

/lgtm
/approve

**Pull request merged**